### DISCUSSION TITLE
How to have a paginated list within a server component layout

### DISCUSSION THREAD

**Discussion by IGassmann** [vercel/next.js#47751] [Help]
### Summary

Take the following example:

![image](https://user-images.githubusercontent.com/4291707/229161522-5e2f4beb-7f26-48a9-bd48-ffa72c26a186.png)

On the left, I have an infinitely scrollable list of items, and on the right, I have the selected item's content. The content on the right and the URL segment update whenever the user clicks on a different item on that list.

**How can I make the list on the left infinitely scrollable while using a React Server Component?**

I thought of using a search parameter such as `https://example.com/items/4?page=5`, but a server layout component cannot access search parameters.

**Comment by mdugue** [comment]
Any ideas here? I was wondering the same, but would ideally like to implement infinite scrolling without hints in the URL. 
My first impression is that it might not be possible at all with RSC right now, but maybe I missed something?
